apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java

```diff
@@ -1568,37 +1568,24 @@ public static Bundle addResourcesToBundle(Bundle bundle, List<IBaseResource> res
             && !Strings.isNullOrEmpty(coverage.getBeneficiary().getReference())) {
           String reference = coverage.getBeneficiary().getReference().replace("Patient/", "");
           if (!Strings.isNullOrEmpty(reference)) {
-            beneIds.add(reference);
+            addBeneIdsToSet(beneIds, reference);
           }
         }
       }
     }
 
-    logBeneIdToMdc(beneIds);
+    LoggingUtils.logBeneIdToMdc(beneIds.stream().toArray(Long[]::new));
 
     return bundle;
   }
 
-  /**
-   * Output list of benefificiary IDs to MDC logging
-   *
-   * @param beneIds the {@link Collection} beneficiary IDs to log
-   */
-  public static void logBeneIdToMdc(Collection<String> beneIds) {
-    if (!beneIds.isEmpty()) {
-      BfdMDC.put("bene_id", String.join(", ", beneIds));
-    }
-  }
-
-  /**
-   * Output beneficiary ID to the MDC logging
-   *
-   * @param beneId the {@link Long} beneficiary ID to log
-   */
-  public static void logBeneIdToMdc(Long beneId) {
-    if (beneId != null) {
-      BfdMDC.put("bene_id", String.valueOf(beneId));
+  private static boolean addBeneIdsToSet(Set<Long> beneIds, String beneId) {
+    try {
+      return beneIds.add(Long.parseLong(beneId));
+    } catch (NumberFormatException e) {
+      LOGGER.warn("Could not parse long from bene_id: " + beneId);
     }
+    return false;
```

Comment:
we can get rid of this and just have the method return void

Reply:
Fixing